services/ui-src/src/views/Profile/Profile.tsx

```diff
@@ -5,16 +5,9 @@ import { Button, Flex, Heading, Text } from "@chakra-ui/react";
 import { useUser } from "utils/auth";
 import { UserRoles } from "utils/types/types";
 
-const userDetails = () => {
-  const userInfo = useUser();
-  const { email, given_name, family_name } = userInfo.user.attributes;
-  const { userRole } = userInfo;
-  const state = userInfo.user.attributes?.["custom:cms_state"] || "";
-  return { email, given_name, family_name, userRole, state };
-};
-
 export const Profile = () => {
-  const { email, given_name, family_name, userRole, state } = userDetails();
+  const { user } = useUser();
+  const { email, given_name, family_name, userRole, state } = user!;
```

Comment:
Prefer

```
const { email, given_name, family_name, userRole, state } = user ?? {};
```

over the ! post-fix. Using the ! post-fix will force the type compiler to assume this is never undefined, and thus the types of required properties like email will be typed as string and not possibly undefined. This can lead to unexpected runtime errors as the type system won't know email could be undefined and will let you run unchecked methods on it. Using `?? {}` instead will ensure that `email` will be typed as `string | undefined` and thus require handing / checking before calling string methods on it.